src/ngp_algorithms/SSTMaxLengthScaleDriver.C

```diff
@@ -64,7 +64,8 @@ void SSTMaxLengthScaleDriver::post_work()
     const unsigned nComponents = 1;
     realm_.periodic_field_update(maxLengthScale, nComponents);
   }
-
+  ngpMaxLengthScale.modify_on_host();
```

Comment:
Is this because realm.periodic_field_update modified on host? I forgot why we can't do the ngp periodic_field_update yet... In any case, shouldn't this be inside the if-statement if(realm.has_periodic)?

Reply:
@alanw0 We need a forced push to device because `doFinalSyncToDevice = false` (in Line 57). The reason to do it this way is because we do `parallel_sum` followed by `periodic_field_update`. Then the operations would be:

```
- Sync from device to host
- Perform parallel_sum
- Sync from host to device 
- If (periodic)
  - Sync from device to host
  - perform periodic exchange
  - Sync from host to device
```

With the above logic we eliminate the intermediate `host -> device -- device -> host` round-trip. It is not too bad here, but there are situations where we do 

```
parallel_sum
copy_owned_to_shared
periodic_field_update
parallel_communicate_field(nonConformalGhosting_)
parallel_communicate_field(oversetGhosting_)
```

Reply:
I agree we don't want the intermediate copies back and forth. Can we switch to using the ngp periodic_field_update? I forgot why we couldn't switch to that.

Reply:
It would be nice if `ngp::Mesh/ngp::Field` could provide a 

```C++
bool fromDevice = true;
bool fromHost = false;
ngp::field_sync_begin(fromDevice, fromHost);
{
  parallel_sum
  copy_owned_to_shared
  parallel_communicate_field(ghosting)
}
ngp::field_sync_end()
```

Reply:
I'm not clear on what would happen within field_sync_begin and field_sync_end. Currently, parallel_sum etc bring the field up from device only if needed, and put it back to device only if directed by the flag. I think that allows for minimal copies. What scenario am I missing?

Reply:
@alanw0 It would parallel `bulk.modification_begin() / bulk.modification_end()` and do the necessary synchronizations. It is just a short-hand for remembering to toggle flags... but yes, it doesn't have anything extra than what we can do with manual passing of flags to individual functions. 

I assume you mean `ngp_apply_constraints`? If so, yes, in this case it will work. However, in situations where we have extra ghostings to handle, we will have to add another flag to prevent `ngp_apply_constraints` from syncing back to device.